src/agoradatatools/etl/transform/model_details.py

```diff
@@ -0,0 +1,190 @@
+
+        # Process biomarkers data
+        model_biomarkers = []
+        model_biomarkers_df = biomarkers_df[biomarkers_df["model"] == model_name]
+
+        # Group biomarkers by evidence_type, tissue, and age_death
+        for (evidence_type, tissue, age), group in model_biomarkers_df.groupby(
+            ["evidence_type", "tissue", "age_death"]
+        ):
+            biomarker_entry = {
+                "model": model_name,
+                "evidence_type": evidence_type,
+                "tissue": tissue,
+                "age": f"{age} months",
+                "units": group["units"].iloc[0],
+                "data": [],
+            }
+
+            # Add individual data points
+            for _, row in group.iterrows():
+                data_point = {
+                    "genotype": row["genotype"],
+                    "sex": row["sex"],
+                    "individual_id": row["individual_id"],
+                    "value": row["value"],
+                }
+                biomarker_entry["data"].append(data_point)
+
+            model_biomarkers.append(biomarker_entry)
+
+        # Process pathology data
+        model_pathology = []
+        model_pathology_df = pathology_df[pathology_df["model"] == model_name]
+
+        # Group pathology by evidence_type, tissue, and age_death
+        for (evidence_type, tissue, age), group in model_pathology_df.groupby(
+            ["evidence_type", "tissue", "age_death"]
+        ):
+            pathology_entry = {
+                "model": model_name,
+                "evidence_type": evidence_type,
+                "tissue": tissue,
+                "age": f"{age} months",
+                "units": group["units"].iloc[0],
+                "data": [],
+            }
+
+            # Add individual data points
+            for _, row in group.iterrows():
+                data_point = {
+                    "genotype": row["genotype"],
+                    "sex": row["sex"],
+                    "individual_id": row["individual_id"],
+                    "value": row["value"],
+                }
+                pathology_entry["data"].append(data_point)
+
+            model_pathology.append(pathology_entry)
+
+        # Build the complete model entry
+        model_entry = {
+            "model": model_name,
+            "matched_controls": model_row["matched_controls"],
+            "model_type": model_row["model_type"],
+            "contributing_group": model_row["contributing_group"],
+            "study_synid": model_row["study_synid"],
+            "rrid": model_row["rrid"],
+            "jax_id": model_row["jax_id"],
+            "alzforum_id": model_row["alzforum_id"],
+            "genotype": model_row["genotype"],
+            "aliases": model_row["aliases"],
+            "genetic_info": genetic_info,
+            "biomarkers": model_biomarkers,
+            "pathology": model_pathology,
+        }
+
+        result.append(model_entry)
```

Comment:
Since this is the only entity within the list, I would just do

```suggestion
        result = [
            {
            "model": model_name,
            "matched_controls": model_row["matched_controls"],
            "model_type": model_row["model_type"],
            "contributing_group": model_row["contributing_group"],
            "study_synid": model_row["study_synid"],
            "rrid": model_row["rrid"],
            "jax_id": model_row["jax_id"],
            "alzforum_id": model_row["alzforum_id"],
            "genotype": model_row["genotype"],
            "aliases": model_row["aliases"],
            "genetic_info": genetic_info,
            "biomarkers": model_biomarkers,
            "pathology": model_pathology,
            }
        ]
```

Reply:
I think there is more than one model in the real dataset.

Reply:
Oh I see I misread the indenting.